OutOfSchool/OutOfSchool.WebApi.Tests/Services/ProviderServicesTests/ProviderTestsHelper.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Threading.Tasks;
+using Moq;
+using OutOfSchool.Services.Models;
+using OutOfSchool.Services.Repository;
+
+namespace OutOfSchool.WebApi.Tests.Services.ProviderServicesTests;
+
+public static class ProviderTestsHelper
+{
+    public static Mock<IProviderRepository> CreateProvidersRepositoryMock(IEnumerable<Provider> providersCollection)
+    {
+        var providersRepository = new Mock<IProviderRepository>();
+        var userExistsResult = false;
+
+        bool UserExist(string userId)
+        {
+            userExistsResult = providersCollection.Any(p => p.UserId.Equals(userId));
+            return userExistsResult;
+        }
+
+        providersRepository.Setup(r => r.ExistsUserId(It.IsAny<string>()))
+            .Callback<string>(user => UserExist(user))
+            .Returns(() => userExistsResult);
```

Comment:
(Readablitiy) I believe this method mock could be simplified to (see [Moq/wiki/Quickstart](https://github.com/devlooped/moq/wiki/Quickstart#methods)):
```
providersRepository.Setup(r => r.ExistsUserId(It.IsAny<string>()))
            .Returns(user => UserExist(user));
```